src/main/java/net/snowflake/client/jdbc/SnowflakeBaseResultSet.java

```diff
@@ -302,7 +302,8 @@ public int findColumn(String columnLabel) throws SQLException {
   public Reader getCharacterStream(int columnIndex) throws SQLException {
     logger.debug("public Reader getCharacterStream(int columnIndex)");
     raiseSQLExceptionIfResultSetIsClosed();
-    return new StringReader(getString(columnIndex));
+    String streamData = getString(columnIndex);
+    return (streamData == null) ? null : new StringReader(streamData);
```

Comment:
Do they expect to have `null` or an empty reader?

Reply:
They expect null. The [documentation|https://docs.oracle.com/javase/7/docs/api/java/sql/ResultSet.html#getCharacterStream(int)] says "a java.io.Reader object that contains the column value; if the value is SQL NULL, the value returned is null in the Java programming language."